### ISSUE TITLE
Circle drawing

### ISSUE THREAD

**Issue by MckinleyX** [xournalpp/xournalpp#3151]
A nice feature to have would to be able to draw circles based on:

- Diameter (start pen at A, end on B, result is a circle with diameter AB)
- Center & radius (start pen on A, end on B, result is a circle with center A and radius AB)

The current situation with the "try to judge where the top left corner will be by eye" is a bit weird.

Thanks!

**Comment by rolandlo** [comment]
The current situation has an advantage for snapping though. 
What you suggest might be implemented via a plugin by what is described in [this suggestion](https://github.com/xournalpp/xournalpp/pull/3025#issuecomment-826806191). Some work will be required to extend the plugin system.

**Comment by robinp7720** [comment]
> The current situation has an advantage for snapping though.
> What you suggest might be implemented via a plugin by what is described in [this suggestion](https://github.com/xournalpp/xournalpp/pull/3025#issuecomment-826806191). Some work will be required to extend the plugin system.

Isn't the suggestion already possible? I had to send my laptop away for repair but iirc if you enable the "Drawing Tools Set Modifiers by Draw Direction" option you can draw circles, squares etc with different properties without needing the keyboard.

Diagonally up, left draws a circle with radius and center. Down, right draws a circle with corner and diameter.

I say the proposed idea has already been implemented.

**Comment by rolandlo** [comment]
> > The current situation has an advantage for snapping though.
> > What you suggest might be implemented via a plugin by what is described in [this suggestion](https://github.com/xournalpp/xournalpp/pull/3025#issuecomment-826806191). Some work will be required to extend the plugin system.
> 
> Isn't the suggestion already possible? I had to send my laptop away for repair but iirc if you enable the "Drawing Tools Set Modifiers by Draw Direction" option you can draw circles, squares etc with different properties without needing the keyboard.
> 
> Diagonally up, left draws a circle with radius and center. Down, right draws a circle with corner and diameter.
> 
> I say the proposed idea has already been implemented.

You are right about drawing a circle with a given center and radius (drawing direction left up or use of ctrl+shift keys). That changed in PR #2707.
You are not right about drawing a circle with a given diameter though. 
It's

**Comment by Tesla-Renting** [comment]
> > > The current situation has an advantage for snapping though.
> > > What you suggest might be implemented via a plugin by what is described in [this suggestion](https://github.com/xournalpp/xournalpp/pull/3025#issuecomment-826806191). Some work will be required to extend the plugin system.
> > 
> > 
> > Isn't the suggestion already possible? I had to send my laptop away for repair but iirc if you enable the "Drawing Tools Set Modifiers by Draw Direction" option you can draw circles, squares etc with different properties without needing the keyboard.
> > Diagonally up, left draws a circle with radius and center. Down, right draws a circle with corner and diameter.
> > I say the proposed idea has already been implemented.
> 
> You are right about drawing a circle with a given center and radius (drawing direction left up or use of ctrl+shift keys). That changed in PR #2707.
> You are not right about drawing a circle with a given diameter though.
> It's

True. Why don't we just make some new subentries below "Draw Ellipse" like

- Draw Ellipse -> corner to corner (fit into rectangle) [currently the default selection]
- Draw Ellipse -> Circle -> corner to projected corner min(dx, dy) [always fits in the spanned rectangle]
- Draw Ellipse -> Circle -> corner to projected corner max(dx, dy) [always greater than the spanned rectangle, currently selected by holding shift]
- Draw Ellipse -> Circle -> center with radius [currently selected by holding control and shift]
- Draw Ellipse -> Circle -> boundary point with diameter

Maybe we have to add the subentries under "Tools", also. Is there any better proposal?
